pkg/kube/namespace/filter.go

```diff
@@ -60,25 +61,26 @@ func NewDiscoveryNamespacesFilter(
 	namespaces.AddEventHandler(controllers.EventHandler[*corev1.Namespace]{
 		AddFunc: func(ns *corev1.Namespace) {
 			f.lock.Lock()
-			defer f.lock.Unlock()
+			created := f.namespaceCreatedLocked(ns.ObjectMeta)
+			f.lock.Unlock()
 			// In rare cases, a namespace may be created after objects in the namespace, because there is no synchronization between watches
 			// So we need to notify if we started selecting namespace
-			if f.namespaceCreatedLocked(ns.ObjectMeta) {
-				f.notifyHandlersLocked(sets.New(ns.Name), nil)
+			if created {
+				f.notifyHandlers(sets.New(ns.Name), nil)
 			}
 		},
 		UpdateFunc: func(old, new *corev1.Namespace) {
 			f.lock.Lock()
-			defer f.lock.Unlock()
 			membershipChanged, namespaceAdded := f.namespaceUpdatedLocked(old.ObjectMeta, new.ObjectMeta)
```

Comment:
change `namespaceUpdatedLocked` name as well

Reply:
I am not sure I understand. The "locked" naming means "this function should be called with the lock held" which remains true

Reply:
While i understand the viceverse

Reply:
ok can we address the renaming in a separate PR? since this will need to be backported, good to keep it minimal?

Reply:
ok